src/modules/commander/HomePosition.cpp

```diff
@@ -319,12 +333,52 @@ void HomePosition::update(bool set_automatically, bool check_if_changed)
 		_gps_epv = vehicle_gps_position.epv;
 
 		const hrt_abstime now = hrt_absolute_time();
-		const bool time = (now < vehicle_gps_position.timestamp + 1_s);
-		const bool fix = vehicle_gps_position.fix_type >= kHomePositionGPSRequiredFixType;
-		const bool eph = vehicle_gps_position.eph < kHomePositionGPSRequiredEPH;
-		const bool epv = vehicle_gps_position.epv < kHomePositionGPSRequiredEPV;
-		const bool evh = vehicle_gps_position.s_variance_m_s < kHomePositionGPSRequiredEVH;
-		_gps_position_for_home_valid = time && fix && eph && epv && evh;
+		const bool time_valid = now < (vehicle_gps_position.timestamp + 1_s);
+		const bool fix_valid = vehicle_gps_position.fix_type >= kHomePositionGPSRequiredFixType;
+		const bool eph_valid = vehicle_gps_position.eph < kHomePositionGPSRequiredEPH;
+		const bool epv_valid = vehicle_gps_position.epv < kHomePositionGPSRequiredEPV;
+		const bool evh_valid = vehicle_gps_position.s_variance_m_s < kHomePositionGPSRequiredEVH;
+
+		_gps_position_for_home_valid = time_valid && fix_valid && eph_valid && epv_valid && evh_valid;
+
+		if (_gps_position_for_home_valid && _last_gps_timestamp != 0 && _last_baro_timestamp != 0
+		    && _takeoff_time != 0 && now - _takeoff_time < kHomePositionCorrectionTimeWindow) {
+
+			const float gps_alt = static_cast<float>(_gps_alt);
+
+			if (!PX4_ISFINITE(_gps_vel_integral)) {
+				_gps_vel_integral = gps_alt; // initialize the gps-vel-integral at same altitude as gps-pos
+				_baro_gps_static_offset = gps_alt - _lpf_baro.getState();
+			}
+
+			_gps_vel_integral += 1e-6f * (vehicle_gps_position.timestamp - _last_gps_timestamp) * (-vehicle_gps_position.vel_d_m_s);
+
+			// correct baro_alt with offset from GPS alt from when the drift integral was initialized
+			const float baro_alt_corrected = _lpf_baro.getState() + _baro_gps_static_offset;
+			const float gps_alt_error = gps_alt + _baro_gps_home_offset;
```

Comment:
Is this variable really storing the GPS altitude error? It I do the math with substituting `_baro_gps_home_offset`, I get 
`gps_alt_error` = `gps_alt_current` - `gps_alt_on_last_home_correction` + `baro_alt_at_last_home_correction`, so it's "how much the gps altitude has increased since the last home position correction, plus the baro altitude when the last home position correction took place". Can you add a comment explaining it?